From d1ee302a2805a06e1d016a2f6c6c856df5c925b2 Mon Sep 17 00:00:00 2001
From: Phil Sutter <psutter@redhat.com>
Date: Wed, 8 May 2024 22:39:40 +0200
Subject: [PATCH] chain: Validate NFTNL_CHAIN_USE, too

JIRA: https://issues.redhat.com/browse/RHEL-28515
Upstream Status: libnftnl commit 104b83489d96642752e774c59e54e816dee85f26

commit 104b83489d96642752e774c59e54e816dee85f26
Author: Phil Sutter <phil@nwl.cc>
Date:   Thu Mar 14 17:22:14 2024 +0100

    chain: Validate NFTNL_CHAIN_USE, too

    Fixes: 53c0ff324598c ("src: add nft_*_attr_{set|get}_data interface")
    Signed-off-by: Phil Sutter <phil@nwl.cc>

Signed-off-by: Phil Sutter <psutter@redhat.com>
---
 src/chain.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/chain.c b/src/chain.c
index dcfcd04..e0b1eaf 100644
--- a/src/chain.c
+++ b/src/chain.c
@@ -196,6 +196,7 @@ static uint32_t nftnl_chain_validate[NFTNL_CHAIN_MAX + 1] = {
 	[NFTNL_CHAIN_HOOKNUM]	= sizeof(uint32_t),
 	[NFTNL_CHAIN_PRIO]		= sizeof(int32_t),
 	[NFTNL_CHAIN_POLICY]		= sizeof(uint32_t),
+	[NFTNL_CHAIN_USE]		= sizeof(uint32_t),
 	[NFTNL_CHAIN_BYTES]		= sizeof(uint64_t),
 	[NFTNL_CHAIN_PACKETS]	= sizeof(uint64_t),
 	[NFTNL_CHAIN_HANDLE]		= sizeof(uint64_t),
